bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java

```diff
@@ -325,16 +302,16 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
         }
     }
 
-    private AudioStream createTemporaryFile(InputStream inputStream, AudioFormat audioFormat) throws TTSException {
-        File mimicDirectory = new File(OpenHAB.getUserDataFolder(), "mimic");
-        mimicDirectory.mkdir();
+    @Override
+    public String getCacheKey(String text, Voice voice, AudioFormat requestedFormat) {
+        MessageDigest md;
         try {
-            File tempFile = File.createTempFile(UUID.randomUUID().toString(), ".wav", mimicDirectory);
-            tempFile.deleteOnExit();
-            Files.copy(inputStream, tempFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
-            return new AutoDeleteFileAudioStream(tempFile, audioFormat);
-        } catch (AudioException | IOException e) {
-            throw new TTSException("Cannot create temporary audio file", e);
+            md = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            return "nomd5algorithm";
         }
+        byte[] binaryKey = ((text + voice.getUID() + requestedFormat.toString() + config.speakingRate
+                + config.audioVolatility + config.phonemeVolatility).getBytes());
+        return String.format("%032x", new BigInteger(1, md.digest(binaryKey)));
```

Comment:
In GoogleTTS, I saw that you excluded the voice from the hash, probably to have file names not containing only a hash ?
Would you like to do the same here ?

Reply:
I did this in google tts this because the old custom google tts cache made the same thing.
Here, I'm not this confident, as I don't know if the voice name will always be "filename friendly".

Reply:
Ok